src/community/features-templating/features-templating-ows/src/main/java/org/geoserver/featurestemplating/ows/wfs/JSONLDGetFeatureResponse.java

```diff
@@ -85,19 +85,33 @@ protected void write(
     }
 
     // retrieve the jsonld context
-    private JsonNode jsonLdContext(FeatureCollectionResponse featureCollection)
+    private EncodingHints optionsToEncodingHints(FeatureCollectionResponse featureCollection)
             throws ExecutionException {
         List<FeatureCollection> collectionList = featureCollection.getFeature();
+        EncodingHints encodingHints = new EncodingHints();
         List<JsonNode> allContexts = new ArrayList<>();
         for (FeatureCollection collection : collectionList) {
             FeatureTypeInfo fti = helper.getFeatureTypeInfo(collection);
             RootBuilder rootBuilder = configuration.getTemplate(fti, identifier.getOutputFormat());
-            JsonNode node = rootBuilder.getVendorOptions().get(CONTEXT, JsonNode.class);
+            VendorOptions options = rootBuilder.getVendorOptions();
+            JsonNode node = options.get(CONTEXT, JsonNode.class);
             if (node == null) node = rootBuilder.getEncodingHints().get(CONTEXT, JsonNode.class);
             if (node != null) allContexts.add(node);
+            addHintIfNotPresent(encodingHints, options, JSONLD_TYPE, String.class);
+            addHintIfNotPresent(encodingHints, options, COLLECTION_NAME, String.class);
         }
         // merge contexts in case we have more then one
-        return nodesUnion(allContexts);
+        JsonNode finalContext = nodesUnion(allContexts);
+        encodingHints.put(CONTEXT, finalContext);
+        return encodingHints;
+    }
+
+    private void addHintIfNotPresent(
```

Comment:
``addHintIfAbsent`` 

In general, avoiding negations helps readability, using the same naming as collections (.e.g ``Map.putIfAbsent``) helps even more.

Reply:
fixed